File: Nocook/src/main/java/com/kwknocook/app/presenter/MainContract.java
//package com.kwknocook.app.presenter;
//
//
//import android.content.Context;
//
//import com.walhalla.landing.MultiplePagination;
//
//import java.util.List;
//
//public interface MainContract {
//    interface View {
//        void showSideNavigationMenu();
//
//        void hideSideNavigationMenu();
//
//        void showBottomNavigationMenu();
//
//        void hideBottomNavigationMenu();
//
//        void showToolbar();
//
//        void hideToolbar();
//
//        void loadUrlInWebView(String url);
//
//        void setupNavigationMenus(List<CatItem> navMenuItems, List<CatItem> bottomNavMenuItems);
//    }
//
//    interface Presenter {
//        void onNavigationItemSelected(String itemId);
//
//        void onBottomNavigationItemSelected(String itemId);
//
//        void fetchRemoteConfig(Context context);
//    }
//}

File: Royspins/src/main/java/com/royspins/app/presenter/MainContract.kt
package com.royspins.app.presenter

import android.content.Context
import com.walhalla.landing.pagination.CatItem


interface MainContract {
    interface View {
        fun showSideNavigationMenu()

        fun hideSideNavigationMenu()

        fun showBottomNavigationMenu()

        fun hideBottomNavigationMenu()

        fun showToolbar()

        fun hideToolbar()

        fun loadUrlInWebView(url: String)

        fun setupNavigationMenus(navMenuItems: List<CatItem>, bottomNavMenuItems: List<CatItem>)
    }

    interface Presenter {
        fun onNavigationItemSelected(itemId: String?)

        fun onBottomNavigationItemSelected(itemId: String?)

        fun fetchRemoteConfig(context: Context)
        fun destroy()
    }
}
